crates/nu-parser/src/errors.rs

```diff
@@ -442,6 +442,10 @@ pub enum ParseError {
     )]
     NotAConstant(#[label = "Value is not a parse-time constant"] Span),
 
+    #[error("Invalid syntax")] // <detail in <entity>.
+    #[diagnostic()]
+    InvalidSyntax(String, String, #[label("{1} in {0}")] Span),
```

Comment:
Can we narrow the name or document where this variant should be used as a lot of parser errors are invalid syntax. Or could this variant also remove `LabelledError` as a kitchen sink catch-all?

The order `{1} in {0}` feels maybe a bit confusing as something you always have to look up to get right.

```suggestion
    #[error("Invalid literal syntax")] // <detail in <entity>.
    #[diagnostic()]
    InvalidLiteralSyntax(String, String, #[label("{1} in {0} literal")] Span),
```

Reply:
I can certainly fix the order of arguments in the error.  Given the existing granularity of error variants, and my strong desire to not rewrite the whole module, I will scope down this name to something like InvalidLiteral.

Reply:
Yeah sounds good